### PR TITLE
[rllib] Model self loss isn't included in all algorithms

### PR DESCRIPTION
**Pull request by ericl**
Closes https://github.com/ray-project/ray/issues/3494

Some other doc tweaks too.

### CODE REVIEW

File: python/ray/rllib/agents/ddpg/ddpg_policy_graph.py [hunk 1]
```diff
@@ -267,15 +266,15 @@ def __init__(self, observation_space, action_space, config):
         # q network evaluation
         prev_update_ops = set(tf.get_collection(tf.GraphKeys.UPDATE_OPS))
         with tf.variable_scope(Q_SCOPE) as scope:
-            q_t, model = self._build_q_network(self.obs_t, observation_space,
-                                               self.act_t)
+            q_t, self.q_model = self._build_q_network(
```
Comment (richardliaw) [inline python/ray/rllib/agents/ddpg/ddpg_policy_graph.py]: why save these variables?

**Review by richardliaw — APPROVED** [review]

**Comment by ericl** [comment]
jenkins retest this please

File: python/ray/rllib/agents/ddpg/ddpg_policy_graph.py (hunk 1 above)
Comment (ericl) [reply python/ray/rllib/agents/ddpg/ddpg_policy_graph.py]: They are used

**Comment by ericl** [comment]
Jenkins retest this please

**Pull request merged**